builtin/providers/vcd/resource_vcd_vapp.go

```diff
@@ -0,0 +1,394 @@
+
+		if e != nil {
+			return fmt.Errorf("Error: %#v", e)
+		}
+
+		e = vcd_client.OrgVdc.Refresh()
+		if e != nil {
+			return fmt.Errorf("Error: %#v", e)
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// err = resource.Retry(4*time.Minute, func() error {
+	// 	err = vcd_client.OrgVdc.InstantiateVAppTemplate(createvapp)
+	//
+	// 	if err != nil {
+	// 		return fmt.Errorf("Error: %#v", err)
+	// 	}
+	// 	return nil
+	// })
+	// if err != nil {
+	// 	return err
+	// }
+
+	vapp, err := vcd_client.OrgVdc.FindVAppByName(d.Get("name").(string))
+
+	err = retryCall(4, func() error {
+		task, err := vapp.ChangeMemorySize(d.Get("memory").(int))
+		if err != nil {
+			return fmt.Errorf("Error changing memory size: %#v", err)
+		}
+
+		return task.WaitTaskCompletion()
+	})
+	if err != nil {
+		return err
+	}
+
+	err = retryCall(4, func() error {
+		task, err := vapp.ChangeCPUcount(d.Get("cpus").(int))
+		if err != nil {
+			return fmt.Errorf("Error changing cpu count: %#v", err)
+		}
+
+		return task.WaitTaskCompletion()
+	})
+	if err != nil {
+		return fmt.Errorf("Error completing task: %#v", err)
+	}
+
+	err = retryCall(4, func() error {
+		task, err := vapp.ChangeVMName(d.Get("name").(string))
+		if err != nil {
+			return fmt.Errorf("Error with vm name change: %#v", err)
+		}
+
+		return task.WaitTaskCompletion()
+	})
+	if err != nil {
+		return fmt.Errorf("Error changing vmname: %#v", err)
+	}
+
+	err = retryCall(4, func() error {
+		task, err := vapp.ChangeNetworkConfig(d.Get("network_name").(string), d.Get("ip").(string))
+		if err != nil {
+			return fmt.Errorf("Error with Networking change: %#v", err)
+		}
+		return task.WaitTaskCompletion()
+	})
+	if err != nil {
+		return fmt.Errorf("Error changing network: %#v", err)
+	}
+
+	err = retryCall(4, func() error {
+		metadata := d.Get("metadata").(map[string]interface{})
+		for k, v := range metadata {
+			task, err := vapp.AddMetadata(k, v.(string))
+			if err != nil {
+				return fmt.Errorf("Error adding metadata: %#v", err)
+			}
+			err = task.WaitTaskCompletion()
+			if err != nil {
+				return fmt.Errorf("Error completing tasks: %#v", err)
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("Error adding metadata: %#v", err)
+	}
+
+	if initscript, ok := d.GetOk("initscript"); ok {
+		err = retryCall(4, func() error {
+			task, err := vapp.RunCustomizationScript(d.Get("name").(string), initscript.(string))
+			if err != nil {
+				return fmt.Errorf("Error with setting init script: %#v", err)
+			}
+			return task.WaitTaskCompletion()
+		})
+		if err != nil {
+			return fmt.Errorf("Error completing tasks: %#v", err)
+		}
+	}
+
+	if d.Get("power_on").(bool) {
+		err = retryCall(4, func() error {
+			task, err := vapp.PowerOn()
+			if err != nil {
+				return fmt.Errorf("Error Powering Up: %#v", err)
+			}
+			return task.WaitTaskCompletion()
+		})
+		if err != nil {
+			return fmt.Errorf("Error completing tasks: %#v", err)
+		}
+	}
```

Comment:
Is it possible to just lean on the implementation in `Update` for these? I believe `HasChange` will be true in the Create case, so you can literally just call your update function. This is what we do, for example, in `aws_instance` implementation, as well as many other places.